<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1022</width>
    <height>603</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Control panel</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/ic_launcher.png</normaloff>:/ic_launcher.png</iconset>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Rounded</enum>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="ControllerAxisWidget" name="tab_altitude">
       <attribute name="title">
        <string>Altitude</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_16"/>
      </widget>
      <widget class="ControllerAxisWidget" name="tab_roll">
       <attribute name="title">
        <string>Roll</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_17"/>
      </widget>
      <widget class="ControllerAxisWidget" name="tab_pitch">
       <attribute name="title">
        <string>Pitch</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_18"/>
      </widget>
      <widget class="ControllerAxisWidget" name="tab_yaw_rate">
       <attribute name="title">
        <string>Yaw rate</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_19"/>
      </widget>
      <widget class="QWidget" name="tab_settings">
       <attribute name="title">
        <string>Settings</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_20">
        <item row="0" column="0">
         <layout class="QFormLayout" name="sensor_config_form">
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <item row="3" column="0" colspan="2">
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>Security</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="1" column="1">
              <widget class="QLineEdit" name="inclinaison_max"/>
             </item>
             <item row="1" column="0">
              <widget class="QCheckBox" name="inclinaison_max_enabled">
               <property name="text">
                <string>Max inclinaison</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="altitude_max"/>
             </item>
             <item row="0" column="0">
              <widget class="QCheckBox" name="altitude_max_enabled">
               <property name="text">
                <string>Max altitude</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="yaw_rate_max_enabled">
               <property name="text">
                <string>Max yaw rate</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="yaw_rate_max"/>
             </item>
            </layout>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QGroupBox" name="groupBox">
            <property name="title">
             <string>Sensors</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLabel" name="accel_low_pass_label">
                 <property name="text">
                  <string>Accel Low pass constant</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="accel_low_pass"/>
               </item>
              </layout>
             </item>
             <item row="1" column="0">
              <layout class="QGridLayout" name="gridLayout_5">
               <item row="0" column="0">
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>Gyro roll bias</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="1">
                <widget class="QLineEdit" name="gyro_roll_bias"/>
               </item>
               <item row="1" column="1">
                <widget class="QLineEdit" name="gyro_pitch_bias"/>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>Gyro yaw bias</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QPushButton" name="calibrate_gyro_bias">
                 <property name="text">
                  <string>Calibrate</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="label_2">
                 <property name="text">
                  <string>Gyro pitch bias</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="6">
                <widget class="QPushButton" name="calibrate_accel">
                 <property name="text">
                  <string>Calibrate</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="4">
                <widget class="QLabel" name="label_6">
                 <property name="text">
                  <string>Accel roll bias</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QLineEdit" name="gyro_yaw_bias"/>
               </item>
               <item row="0" column="6">
                <widget class="QLineEdit" name="accel_roll_bias"/>
               </item>
               <item row="1" column="4">
                <widget class="QLabel" name="label_7">
                 <property name="text">
                  <string>Accel pitch bias</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="6">
                <widget class="QLineEdit" name="accel_pitch_bias"/>
               </item>
               <item row="0" column="2">
                <widget class="QLabel" name="label_8">
                 <property name="text">
                  <string>Gyro roll gain</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="2">
                <widget class="QLabel" name="label_9">
                 <property name="text">
                  <string>Gyro pitch gain</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="2">
                <widget class="QLabel" name="label_10">
                 <property name="text">
                  <string>Gyro yaw gain</string>
                 </property>
                </widget>
               </item>
               <item row="0" column="3">
                <widget class="QLineEdit" name="gyro_roll_gain"/>
               </item>
               <item row="1" column="3">
                <widget class="QLineEdit" name="gyro_pitch_gain"/>
               </item>
               <item row="2" column="3">
                <widget class="QLineEdit" name="gyro_yaw_gain"/>
               </item>
               <item row="3" column="3">
                <widget class="QPushButton" name="calibrate_gyro_gain">
                 <property name="text">
                  <string>Calibrate</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="apply_modulo">
               <property name="text">
                <string>Apply a modulo 2*PI to the angles</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>Motors</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_4">
             <item row="0" column="0">
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>Minimum PWM value</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="min_pwm"/>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>Maximum PWM value</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="max_pwm"/>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1022</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Connect"/>
    <addaction name="action_Disconnect"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="action_Restore"/>
    <addaction name="separator"/>
    <addaction name="action_Quit"/>
   </widget>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="action_About"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Help"/>
  </widget>
  <action name="action_Connect">
   <property name="text">
    <string>&amp;Connect...</string>
   </property>
  </action>
  <action name="action_Quit">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="action_Disconnect">
   <property name="text">
    <string>&amp;Disconnect</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="action_Restore">
   <property name="text">
    <string>&amp;Restore</string>
   </property>
  </action>
  <action name="actionPause">
   <property name="text">
    <string>Pause</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ControllerAxisWidget</class>
   <extends>QWidget</extends>
   <header>controlleraxiswidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>action_Quit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>510</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
